OpenSim/Common/Component.cpp

```diff
@@ -820,9 +820,9 @@ setDiscreteVariableValue(SimTK::State& s, const std::string& name, double value)
     }
 }
 
-void Component::constructOutputForStateVariable(const std::string& name)
+int Component::constructOutputForStateVariable(const std::string& name)
```

Comment:
Should we settle now to use `SimTK_DEFINE_UNIQUE_INDEX_TYPE()`? instead of an int?
With the potential of having many indices I like using the unique index to thwart accidentally mixing indices up.